libsupport/include/katana/EntityTypeManager.h

```diff
@@ -194,15 +198,106 @@ class EntityTypeManager {
     return (sub_atomic_types & super_atomic_types) == sub_atomic_types;
   }
 
+  const EntityTypeIDToSetOfEntityTypeIDsMap&
+  GetEntityTypeIDToAtomicEntityTypeIDs() const {
+    return entity_type_id_to_atomic_entity_type_ids_;
+  }
+
+  const EntityTypeIDToAtomicTypeNameMap& GetEntityTypeIDToAtomicTypeNameMap()
+      const {
+    return atomic_entity_type_id_to_type_name_;
+  }
```

Comment:
These should NOT be exposed. The leak representation information.

Reply:
we have to be able to pull the information out to convert a manager to what we store. We could instead have EntityTypeManager do the conversion, but I'm not sure that is appropriate since EntityTypeManager shouldn't have to care about how we store its data.

Reply:
OK. I see the issue. I'll look at it later. This is one case where I think friend classse really do make sense.

Reply:
definitely open to suggestions for better ways to do this in a future PR